rayon-core/src/thread_pool/test.rs

```diff
@@ -57,3 +58,22 @@ fn sleeper_stop() {
     // should lead to worker threads stopping
     registry.wait_until_stopped();
 }
+
+#[test]
+fn failed_thread_stack() {
+    let config = Configuration::new().set_stack_size(::std::usize::MAX);
+    let pool = ThreadPool::new(config);
+    assert!(pool.is_err(), "thread stack should have failed!");
+    // TODO how to make sure threads actually stopped?
+}
+
+#[test]
+fn panic_thread_name() {
+    let config = Configuration::new().set_num_threads(10).set_thread_name(|i| {
+        if i == 5 { panic!(); }
+        format!("panic_thread_name#{}", i)
+    });
+    let pool = unwind::halt_unwinding(|| ThreadPool::new(config));
+    assert!(pool.is_err(), "thread-name panic should propagate!");
+    // TODO how to make sure threads actually stopped?
```

Comment:
Once #271 lands, I could use callbacks to count the actual thread starts and exits.